### ISSUE TITLE
It is netdata instalation hacked ?

### ISSUE THREAD

**Issue by irekpias** [netdata/netdata#5207]
Hi. 
I use netdata some time. In this day/night i want update my netdata instalation, so i run: ./netdata-updater.sh He grabs some files - updated netdata and run again netdata. But there is no more uninstall netdata script and script ./netdata-updater has 0 length. So further updates aren't possible any more. What is going on ?? I have many netdata installations and now they all are broken. IS netdata hacked, or so ?

I make fresh installation and my version is: v1.12.0-rc3-3-g711dbd76 but after running script: ./netdata-updater.sh i have uknow version of nedata (he grabs this file after running netdata: https://storage.googleapis.com/netdata-nightlies/netdata-latest.tar.gz and compile, after that ./netdata-updater.sh has 0 lenght.

**Comment by irekpias** [comment]
clean install of netdata give me this:
![image](https://user-images.githubusercontent.com/46800639/51359244-21d60900-1ac7-11e9-9761-273325bd9f8c.png)

But after run script: ./netdata-updater from cron.daily i have this:
![image](https://user-images.githubusercontent.com/46800639/51359290-5c3fa600-1ac7-11e9-96d6-637d28cc98f2.png)

What a mess !

**Comment by WoLpH** [comment]
I've noticed the same while debugging this issue: https://github.com/netdata/netdata/issues/5180

It seems that running the update script kills the update script after and I haven't found a clean way to fix it yet. Reinstalling with the `--auto-update` flag gets the update script working again but as soon as the daily update cron job starts it dies again.

**Comment by WoLpH** [comment]
The specific output from the update command which kills it:
```
Uninstall script is located at: ./netdata-uninstaller.sh
 --- Installing new netdata-updater in cron ---
sed: can't read ./packaging/installer/netdata-updater.sh: No such file or directory
Update script is located at /etc/cron.daily/netdata-updater
```

**Comment by WoLpH** [comment]
I'm guessing this commit could be the culprit: https://github.com/netdata/netdata/commit/d5be47e4760354cbdfccdc2740b7ff091daaf2f9

I've noticed the bug on 2 separate Ubuntu 16.04 hosts, not a 100% sure it's not environment specific but it doesn't appear to be.

**Comment by paulfantom** [comment]
No, netdata wasn't "hacked". We are in a process of changing how uninstallation and automatic updates work to make them more robust and independent from git repository. That's why we are using travis CI to create nightly packages and put them into Google Cloud Storage. Also we changed how our versioning works, so netdata can be build without git, for this to work we are storing latest version number in https://github.com/netdata/netdata/blob/master/packaging/version. As of now latest version is `v1.12.0-rc2-51-nightly` and it is the same as `v1.12.0-rc2-51-3ae4c18`. Difference is that first was built without git and second with git.

Unfortunately it also broke version check in GUI (issue: #5204).


The update process was always meant to be an automatic thing enabled during installation with `--auto-update` flag, so we removed creation of `netdata-updater.sh` script from source code dir and we are putting it directly inside cron directory (instead of previously putting there a symlink).

Unfortunately it seems that our testing was insufficient and we haven't caught this bug. Fix is already on the way in #5209.

I am very sorry that it broke your installations.

As of uninstaller, we also changed how it works, so we have an uninstaller which is self-container, can be downloaded directly from https://github.com/netdata/netdata/blob/master/packaging/installer/netdata-uninstaller.sh and run. You can read more about it in our docs: https://docs.netdata.cloud/packaging/installer/uninstall/
